src/sc_data/data.py

```diff
@@ -74,7 +74,7 @@ def update(self):
             and (db_hash := r.headers.get("x-amz-meta-hash", time.time()))
             != self.actual_db_hash
         ):
-            tmpfile = tempfile.NamedTemporaryFile()
+            tmpfile = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
This will leave the temp files around. I'm not familiar with this on Windows, what problem does this try to solve?

Reply:
https://stackoverflow.com/a/15590253/564164

Reply:
maybe as per https://github.com/python/cpython/issues/58451, and it's resolution at https://github.com/python/cpython/issues/58451, `delete_on_close` might help? :man_shrugging:

Reply:
no, with `delete_on_close`, we get:

> The process cannot access the file because it is being used by another process

Reply:
> This will leave the temp files around.

wait, `close_tmpfiles` will not remove those?

Reply:
FTR the related docs clearly describe how Win behaves compare to posix https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

Reply:
> > This will leave the temp files around.
> 
> wait, `close_tmpfiles` will not remove those? Or you mean one (the most recent)? well, I hope you can find a better solution for this 🙏

No, it relies on the delete on garbage collect behaviour. The problem with this is that file should be auto-deleted in most of the cases, but if we disable that, we would have to implement the same (like an atexit handler or similar).
I didn't think we target Windows. :-O

Reply:
> The problem with this is that file should be auto-deleted in most of the cases, but if we disable that, we would have to implement the same (like an atexit handler or similar).

`close_tmpfiles` is already called on old files + registered at `safe_exit`, so would not be enough to change the `.close` calls to unlink?

> I didn't think we target Windows. :-O

well, @Palabola is on windows, and he is working on the keeper, so now we do 🙃

Reply:
That should do, changed.
WSL? :)